### PR TITLE
OSSMDOC-633: Document preliminary integration with cert-manager.

### PR DESCRIPTION
**Pull request by JStickler**
Version(s):  main only, no cherry pick to any branch (DRAFT)

Issue: [OSSMDOC-633](https://issues.redhat.com/browse/OSSMDOC-633)

This PR adds a new DRAFT module with documentation for preliminary integration with cert-manager.  It also changes an existing heading that referred to cert-manager but the procedure does not actually use cert-manager.

UPDATE - After meeting with Engineering and QE on October 14, it was determined that this feature had a bug and might not be ready in time for the next release. Engineering said that the draft of the documentation was good for the current state of the code. Because Julie is changing jobs and won't be around to finish the work, I am checking in the new module so that the writer who picks up this work when the feature is ready has access to the draft.  Merging to main only, not included in any assembly so not published.  Got permission from the DPM to do so.

### CODE REVIEW

File: modules/ossm-config-external-ca.adoc [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    organizations:
+      - cluster.local
+      - cert-manager
+  issuerRef:
+    name: selfsigned
+    kind: Issuer
+    group: cert-manager.io
+----
+
+//Not sure what to title this example?  Or even if we need to include it?
```
Comment (rcernich) [inline modules/ossm-config-external-ca.adoc:65]: Yes, this generates an `Issuer` named `istio-ca`  that uses the key/cert generated above (the `Certificate`).  I also think we need to include the `selfsigned` `Issuer` too, as it's used to create/sign the certificate above.

Of all of this, I think the flow is:
* Root CA - in this case a self-signed certificate generated using `Issuer.spec.selfSigned` (I think we need to add this for the example.)
* Istio CA - This is still the `istio-ca` `Secret` located in the `istio-system` namespace, but it is generated from the `istio-ca` `Certificate` (above).  This certificate will be used to sign certificates for the `istio-ca` `Issuer` configured below.

I *think* this is correct.

Comment (JStickler) [reply modules/ossm-config-external-ca.adoc]: @rcernich I've revised based on my current understanding of the workflow.  But the examples in 
https://github.com/rcernich/istio-operator/blob/f92e8142120a11c2b8ca0487907193fad2d11272/deploy/examples/cert-manager/selfsigned-ca.yaml

And 
https://github.com/rcernich/istio-operator/blob/f92e8142120a11c2b8ca0487907193fad2d11272/deploy/examples/cert-manager/smcp.yaml

Don't have matching values for the secret names.  Should they match?  (See my comment in the example).

**Comment by JStickler** [comment]
Merging to main branch only, as a draft topic for next writer.

**Comment by bergerhoffer** [comment]
@kalexand-rh wdyt of this? I'd probably either rather revert this, or move them to the unused_topics directory. Thoughts?

**Comment by kalexand-rh** [comment]
I think it would be better to revert it. @rh-tokeefe, please note this PR so you can CP it into your branch when you resume this work.

**Comment by gwynnemonahan** [comment]
This PR can be closed as it was a draft PR for code that did not ship. 

The doc PR for the shipping code is here: https://github.com/openshift/openshift-docs/pull/57747

**Pull request merged**